### ISSUE TITLE
bids addpath

### ISSUE THREAD

**Issue by CerenB** [cpp-lln-lab/CPP_BIDS#68]
Atm, when the bids added with 
`    addpath(fullfile(pth, 'lib', 'CPP_BIDS', 'src'));`
due to the subfunc folder is not inside src, would it be an issue? Will check it now. (after resolving the following error)

**Comment by CerenB** [comment]
OK let me rephrase it, it's submodule issue in README file
` git submodule add https://github.com/cpp-lln-lab/CPP_BIDS.git` is not enough to add CPP_BIDS as submodules in my repo. It gives the following error:
```
>> cfg = createFilename(cfg);
Error using checkCppBidsDependencies>checkSubmodule (line 33)
Git submodules are not cloned!Try this in your terminal: git submodule update --recursive

Error in checkCppBidsDependencies (line 16)
        checkSubmodule(fullfile(pth, 'lib', 'JSONio'));

Error in checkCFG (line 11)
    checkCppBidsDependencies(cfg);

Error in createFilename (line 15)
    cfg = checkCFG(cfg);
```

**Comment by Remi-Gau** [comment]
I think you should do this after to actually get the code of the submodule
```
git submodule init
```

**Comment by CerenB** [comment]
that should be added to README ? 
I can do it soon -and/or whoever adds it should close the issue.

**Comment by CerenB** [comment]
@Remi-Gau not sure how did I ended up here but, something not tracked (README file) while I was working on my repo. GitKraken refused to show me my repo. Then I started using the terminal to check what's the issue:
```
$ git checkout fmri_task
error: The following untracked working tree files would be removed by checkout:
	lib/CPP_BIDS/README.md
	lib/CPP_BIDS/tests/test_createFilename.m
Please move or remove them before you switch branches.
Aborting
```
What shall I do here? I do not want to re-install the repo because I have some local changes, not everything is pushed yet to remote.

**Comment by Remi-Gau** [comment]
@CerenB 

- what does `git status` tell you?
- `lib/CPP_BIDS` is a submodule, right?

**Comment by CerenB** [comment]
@Remi-Gau this is interesting, i did couple of things. First git status was telling me there were loads of uncommitted changes. So I committed them, in origin and pushed them remote as well (created new remote branch). Then I was saying:
```
fatal: The current branch bids_submodule has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin bids_submodule
```
So I ran:
`git push --set-upstream origin bids_submodule
`
Now it seems working but README file gave an error in GitKraken says it doesn't exist... Maybe GitKraken is not super good with submodules?

**Comment by Remi-Gau** [comment]
actually gitkraken won't show you the changes in submodule unless you "open" that submodule.

bottom left of the git kraken window and right-click

![Screenshot 2020-08-04 15:18:49](https://user-images.githubusercontent.com/6961185/89298641-dca40880-d665-11ea-8753-e13518cfdb12.png)

**Comment by Remi-Gau** [comment]
so if you do `git status` git will tell you only about the submodule and will pretend the outside world does not exist.

if you do `git status` in a repo where one of the sub module has been modified it will show this:
```
modified:   lib/CPP_PTB (modified content, untracked content)
```
But it won't tell you what the modified content is.

**Comment by CerenB** [comment]
Hmm so it wasn't that. but on gitkraken something went wrong because I was changing branch names etc.. then I had sort of this issue:
https://stackoverflow.com/questions/59021836/gitkraken-doesnt-show-branches-and-commits

I was wondering changing from branches (with/out submodules set) could cause an issue. 
Now the git status is:
```
(base) mac-114-168:RhythmCateg battal$ git status
On branch fmri_task
Your branch is up to date with 'origin/fmri_task'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	stimuli/Grahn2007_complex.txt
	stimuli/Grahn2007_simple.txt

nothing added to commit but untracked files present (use "git add" to track)
```
I can see in Atom that .gitmodules exist and not empty. 

I'll check what would cause "README file cannot be seen/read" next...
